core/src/replay_stage.rs

```diff
@@ -854,7 +854,10 @@ impl ReplayStage {
     ) {
         let start_slot = poh_recorder.lock().unwrap().start_slot();
         if let Some(latest_leader_slot) = progress.get_latest_leader_slot(start_slot) {
-            if !progress.is_propagated(latest_leader_slot) {
+            // We just returned a postivie result above from `get_latest_leader_slot`, so
+            // `start_slot` must exist in the progress map above
+            let must_exist_in_progress_map = true;
+            if !progress.is_propagated(start_slot, must_exist_in_progress_map) {
```

Comment:
@jbiseda I was looking through this code and was wondering if the `is_propagated` check should run on `start_slot` instead of `latest_leader_slot` here. 

The reason being the `is_propagated(start_slot)` function unintuitively checks the propagation status of the latest leader slot of the `start_slot` https://github.com/solana-labs/solana/blob/4fb1a180b9868db33c24cc8cd6cadc8702e9279a/core/src/progress_map.rs#L468-L484, not the propagation status of the `start_slot` itself

Reply:
I think it would be more straightforward if `is_propagated` just operated on the provided slot. If we want the leader status then we get the leader slot and query for the leader. What do you think?

Reply:
yeah agreed